libraries/lib-basic-ui/BasicUI.h

```diff
@@ -315,20 +315,24 @@ inline std::unique_ptr<GenericProgressDialog> MakeGenericProgress(
       return nullptr;
 }
 
-template <typename RangeItemType, typename RangeType>
+template <typename ItType>
 void SplitProgress(
-   ProgressReporter parent, RangeType range,
-   std::function<void(RangeItemType, const ProgressReporter&)> action)
+   ItType first, ItType last,
+   std::function<void(typename ItType::value_type, const ProgressReporter&)>
+      action,
+   ProgressReporter parent)
```

Comment:
`ProgressReporter` too can be a deduced type, allowing more generality and avoiding some more `std::function` constructions.

Reply:
But doesn't the user then loose the information that it takes a double and returns a void just by looking at the function's signature?